geopyspark/geotrellis/catalog.py

```diff
@@ -365,14 +369,14 @@ def query(layer_type,
 
     else:
         if time_intervals:
-            time_intervals = [time.isoformat() for time in time_intervals]
+            time_intervals = [time.isoformat() + "Z" for time in time_intervals]
```

Comment:
Thats not safe way to convert the datetime. `Z` prefix indicates that the timestamp is in UTC timezone, which it may or may not be.

```py
In [38]: dt = datetime.combine(d, t)

In [39]: d = date(2005, 7, 14)

In [40]: dt = datetime.combine(d, t)

In [41]: dt.isoformat()
Out[41]: '2005-07-14T12:30:00'

In [42]: dt.astimezone(pytz.utc).isoformat()
Out[42]: '2005-07-14T16:30:00+00:00'

In [43]: dt.astimezone(eastern).isoformat()
Out[43]: '2005-07-14T12:30:00-04:00'
```

`datetime` instance may already have timezone set. If its not there is an implied system default (EST in my case). Probably the safest way is to convert to UTC time zone. I believe the java parser will accept timestamp with `+00:00` for timezone just fine.